### PR TITLE
Refactor absence, add proper staffing and update migrations

### PR DESCRIPTION
**Pull request by jonasbjoralt**
OBS: You need to reset your DB to the vacation-migration before migrating

`dotnet ef database update VacationAndAbsence`
 Then do latest migrations and update

### CODE REVIEW

File: backend/Core/DomainModels/PlannedAbsence.cs
Comment (yelodevopsi) [inline backend/Core/DomainModels/PlannedAbsence.cs:1]: Need a PK id here

File: backend/Infrastructure/DatabaseContext/ApplicationContext.cs [hunk 1]
```diff
@@ -35,23 +35,47 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
             .WithOne(dept => dept.Organization);
 
         modelBuilder.Entity<Organization>()
-            .HasMany<Client>(organization => organization.Clients)
```
Comment (yelodevopsi) [inline backend/Infrastructure/DatabaseContext/ApplicationContext.cs:38]: Was there are requirements to have Customers/Clients in org from before?
Why was this removed?

File: backend/Infrastructure/DatabaseContext/ApplicationContext.cs [hunk 2]
```diff
@@ -35,23 +35,47 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
...
+            .HasMany<Absence>(organization => organization.AbsenceTypes)
+            .WithOne(absence => absence.Organization);
+
+        modelBuilder.Entity<Organization>()
+            .HasMany<Customer>(organization => organization.Customers)
             .WithOne(client => client.Organization);
 
-        modelBuilder.Entity<Client>()
+        modelBuilder.Entity<Customer>()
             .HasMany<Project>(client => client.Projects)
```
Comment (yelodevopsi) [inline backend/Infrastructure/DatabaseContext/ApplicationContext.cs:46]: Better to rename "client" to "customer" on line 42 here?

File: backend/Core/DomainModels/PlannedAbsence.cs
Comment (jonasbjoralt) [reply backend/Core/DomainModels/PlannedAbsence.cs:1]: done

File: backend/Infrastructure/DatabaseContext/ApplicationContext.cs (hunk 1 above)
Comment (jonasbjoralt) [reply backend/Infrastructure/DatabaseContext/ApplicationContext.cs:38]: Not removed, this line is now mapping absence. Customer is mapped on L41. Git probably marked this as removed since line 37 is identical.

**Review by yelodevopsi — COMMENTED** [review]
Look over/answer Q

**Review by yelodevopsi — APPROVED** [review]
